Switch containers/[image,storage,common] to go.podman.io.

jankaluza opened the pull request:
These go packages were migrated to a monorepo, as stated in the https://blog.podman.io/2025/08/migration-to-the-container-libs-monorepo-is-complete/.

This commit updates the crc package to use these packages from new locations.

It was generated using following commands:

```
$ find . -type f -name '*.go' -exec sed -i -e 's,"github.com/containers/image,"go.podman.io/image,g' {} \;
$ find . -type f -name '*.go' -exec sed -i -e 's,"github.com/containers/common,"go.podman.io/common,g' {} \;
$ find . -type f -name '*.go' -exec sed -i -e 's,"github.com/containers/storage,"go.podman.io/storage,g' {} \;
$ goimports -v -w .
$ git checkout vendor/
$ vi go.mod # to change the storage, image and common imports
$ go mod tidy
$ go mod vendor
```

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Chores**
  * Promoted and updated the shared utilities dependency to a newer provider/version to align packages.
  * Adjusted internal imports to reference the updated shared utilities package; no public APIs or user-facing behavior changed.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

cfergeau commented on pkg/config/config.go:
```diff
@@ -20,7 +20,7 @@ import (
 	"strings"
 
-	"github.com/containers/common/pkg/strongunits"
+	"go.podman.io/common/pkg/strongunits"
```
This comment needs a closer look as podman/crc/… are using this code
/hold

cfergeau commented:
I’ve redone the commit against latest `main` using your instructions and force pushed to your branch.

cfergeau commented:
/ok-to-test

cfergeau committed:
Switch containers/[image,storage,common] to go.podman.io.

These go packages were migrated to a monorepo, as stated in the https://blog.podman.io/2025/08/migration-to-the-container-libs-monorepo-is-complete/.

This commit updates the crc package to use these packages from new locations.

It was generated using following commands:

```
$ find . -type f -name '*.go' -exec sed -i -e 's,"github.com/containers/image,"go.podman.io/image,g' {} \;
$ find . -type f -name '*.go' -exec sed -i -e 's,"github.com/containers/common,"go.podman.io/common,g' {} \;
$ find . -type f -name '*.go' -exec sed -i -e 's,"github.com/containers/storage,"go.podman.io/storage,g' {} \;
$ goimports -v -w .
$ git checkout vendor/
$ vi go.mod # to change the storage, image and common imports
$ go mod tidy
$ go mod vendor
```

Signed-off-by: Jan Kaluza <jkaluza@redhat.com>

vyasgun reviewed (approved):
/approve
/lgtm
